xbmc/utils/ScreenshotSurfaceGL.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ *      Copyright (C) 2015 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "ScreenshotSurfaceGL.h"
+
+#include "guilib/GUIComponent.h"
+#include "guilib/GUIWindowManager.h"
+#include "ServiceBroker.h"
+#include "threads/SingleLock.h"
+#include "windowing/GraphicContext.h"
+
+#include "system_gl.h"
+
+bool CScreenshotSurfaceGL::Capture()
+{
+  CSingleLock lock(CServiceBroker::GetWinSystem()->GetGfxContext());
+  CServiceBroker::GetGUI()->GetWindowManager().Render();
```

Comment:
you should test GetWinSystem and GetGUI for nullptr. utils can't assume that there is a gui or winsystem

Reply:
I didn't really want to change existing code, I just wanted to reorganise it first

Reply:
why don't you want to make improvments that are necessary when you touch this. that adds value, plain refactoring does not